Name clashes in C code when using ref and non-ref actual type parameters

This small example causes name clashes in the C code:
```
 > cat test_1121.fz 
test_1121 =>
  a i32 42
  a Any 47
  a (ref i32) 11

a(T type, v T) =>
  say v

test_1121
 > ./build/bin/fz -c test_1121.fz 
#universe.c:9011:8: error: redefinition of 'fzT_1a_w_i32'
struct fzT_1a_w_i32
       ^
#universe.c:9005:8: note: previous definition is here
struct fzT_1a_w_i32
       ^
#universe.c:9326:6: error: conflicting types for 'fzC_1a_w_i32'
void fzC_1a_w_i32(fzT_1i32 arg0);
     ^
#universe.c:9325:6: note: previous declaration is here
void fzC_1a_w_i32(fzT__R1i32* arg0);
     ^
#universe.c:20726:18: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'fzT__R1i32 *' (aka 'struct fzT__R1i32 *') [-Werror,-Wint-conversion]
    fzC_1a_w_i32(42);
                 ^~
#universe.c:9325:31: note: passing argument to parameter 'arg0' here
void fzC_1a_w_i32(fzT__R1i32* arg0);
                              ^
#universe.c:20793:6: error: conflicting types for 'fzC_1a_w_i32'
void fzC_1a_w_i32(fzT_1i32 arg0)
     ^
#universe.c:20773:6: note: previous definition is here
void fzC_1a_w_i32(fzT__R1i32* arg0)
     ^
#universe.c:20800:19: error: incompatible integer to pointer conversion assigning to 'fzT__R1i32 *' (aka 'struct fzT__R1i32 *') from 'fzT_1i32' (aka 'int') [-Werror,-Wint-conversion]
    fzCur.fzF_0_v = arg0;
                  ^ ~~~~
#universe.c:20808:19: error: incompatible pointer to integer conversion assigning to 'fzT_1i32' (aka 'int') from 'fzT__R1i32 *' (aka 'struct fzT__R1i32 *') [-Werror,-Wint-conversion]
    fzM_0->fields = fzCur.fzF_0_v;
                  ^ ~~~~~~~~~~~~~
6 errors generated.

error 1: C backend: C compiler failed
C compiler call 'clang -Wall -Werror -Wno-gnu-empty-struct -Wno-unused-variable -Wno-unused-label -Wno-unused-but-set-variable -Wno-unused-function -Wno-infinite-recursion -O3 -lm -lpthread -o #universe #universe.c' failed with exit code '1'

one error.
```
These are really only naming problems, they can be overcome by renaming `a` as `abcabcabcabcabcabcabcabc` (which causes the C backend to use an abbreviated name plus a unique id:
```
 > cat test_1121b.fz 
test_1121 =>
  abcabcabcabcabcabcabcabc i32 42
  abcabcabcabcabcabcabcabc Any 47
  abcabcabcabcabcabcabcabc (ref i32) 11

abcabcabcabcabcabcabcabc(T type, v T) =>
  say v

test_1121
 > ./build/bin/fz -c test_1121b.fz 
 > ./#universe 
42
47
11
```
This is related to #1121, the example there runs into this problem after the DFA was fixed.

--- a/src/dev/flang/ast/AbstractType.java
+++ b/src/dev/flang/ast/AbstractType.java
@@ -1248,12 +1248,7 @@ public abstract class AbstractType extends ANY implements Comparable<AbstractTyp
               + featureOfType().featureName().baseName();
         for (var g : generics())
           {
-            var gs = g.asString();
-            if (gs.indexOf(" ") >= 0)
-              {
-                gs = "(" + gs + ")";
-              }
-            result = result + " " + gs;
+            result = result + " " + g.asStringWrapped();
           }
         if (isThisType())
           {
@@ -1264,7 +1259,6 @@ public abstract class AbstractType extends ANY implements Comparable<AbstractTyp
   }
 
 
-
   /**
    * wrap the result of toString in parentheses if necessary
    */
@@ -1273,6 +1267,17 @@ public abstract class AbstractType extends ANY implements Comparable<AbstractTyp
     return toString().contains(" ")
            ? "(" + toString() + ")"
            : toString();
+  }
+
+
+  /**
+   * wrap the result of asString in parentheses if necessary
+   */
+  public String asStringWrapped()
+  {
+    var s = asString();
+    return s.contains(" ") ? "(" + s + ")"
+                           :       s      ;
   }
 
 

--- a/src/dev/flang/fuir/FUIR.java
+++ b/src/dev/flang/fuir/FUIR.java
@@ -504,7 +504,7 @@ public class FUIR extends IR
     var cc = clazz(cl);
     var res = cc.feature().featureName().baseName();
     res = res + cc._type.generics()
-      .toString(" ", " ", "", t -> t.toStringWrapped());
+      .toString(" ", " ", "", t -> t.asStringWrapped());
     return res;
   }
 

